ds3/ds3.py

```diff
@@ -6777,15 +6765,13 @@ class GetObjectResponse(AbstractResponse):
   
   def process_response(self, response):
     self.__check_status_codes__([200, 206])
-    localFile = None
-    if self.request.stream:
-      localFile = self.request.stream
-    else:
-      localFile = open(self.request.effective_file_name, "wb")
-    if self.request.offset:
-      localFile.seek(self.request.offset, 0)
-    localFile.write(response.read())
-    localFile.close()
+    stream = self.request.stream
+    bytes_read = response.read()
+    while bytes_read:
+      stream.write(bytes_read)
+      bytes_read = response.read()
+    stream.write(response.read())
+    stream.close()
```

Comment:
Also, who is responsible for closing the response stream?  One last thing, why the last stream.write(response.read()) call outside of the loop?  Didn't it already finish reading all the data?